Dockerfile

```diff
@@ -0,0 +1,375 @@
+    && curl https://dl.yarnpkg.com/debian/pubkey.gpg | apt-key add - > /dev/null \
+    && echo "deb https://dl.yarnpkg.com/debian/ stable main" > /etc/apt/sources.list.d/yarn.list \
+    # Note missing man directories on debian-buster
+    # https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=863199
+    && mkdir -pv /usr/share/man/man1 \
+    && mkdir -pv /usr/share/man/man7 \
+    && apt-get update \
+    && apt-get install -y --no-install-recommends \
+           apt-transport-https \
+           apt-utils \
+           build-essential \
+           ca-certificates \
+           curl \
+           gnupg \
+           dirmngr \
+           freetds-bin \
+           freetds-dev \
+           gosu \
+           krb5-user \
+           ldap-utils \
+           libffi-dev \
+           libkrb5-dev \
+           libpq-dev \
+           libsasl2-2 \
+           libsasl2-dev \
+           libsasl2-modules \
+           libssl-dev \
+           locales  \
+           lsb-release \
+           nodejs \
+           openssh-client \
+           postgresql-client \
+           python-selinux \
+           sasl2-bin \
+           software-properties-common \
+           sqlite3 \
+           sudo \
+           unixodbc \
+           unixodbc-dev \
+           yarn \
+    && apt-get autoremove -yqq --purge \
+    && apt-get clean \
+    && rm -rf /var/lib/apt/lists/*
+
+# Install MySQL client from Oracle repositories (Debian installs mariadb)
+RUN KEY="A4A9406876FCBD3C456770C88C718D3B5072E1F5" \
+    && GNUPGHOME="$(mktemp -d)" \
+    && export GNUPGHOME \
+    && for KEYSERVER in $(shuf -e \
+            ha.pool.sks-keyservers.net \
+            hkp://p80.pool.sks-keyservers.net:80 \
+            keyserver.ubuntu.com \
+            hkp://keyserver.ubuntu.com:80 \
+            pgp.mit.edu) ; do \
+          gpg --keyserver "${KEYSERVER}" --recv-keys "${KEY}" && break || true ; \
+       done \
+    && gpg --export "${KEY}" | apt-key add - \
+    && gpgconf --kill all \
+    rm -rf "${GNUPGHOME}"; \
+    apt-key list > /dev/null \
+    && echo "deb http://repo.mysql.com/apt/debian/ stretch mysql-5.7" | tee -a /etc/apt/sources.list.d/mysql.list \
+    && apt-get update \
+    && apt-get install --no-install-recommends -y \
+        libmysqlclient-dev \
+        mysql-client \
+    && apt-get autoremove -yqq --purge \
+    && apt-get clean && rm -rf /var/lib/apt/lists/*
+
+# disable bytecode generation (only in build image - it does not affect the main image)
+ENV PYTHONDONTWRITEBYTECODE=1
+
+RUN pip install --upgrade pip==${PIP_VERSION}
+
+# We do not need it here technially but we are going to
+# override COPY_SOURCE with "." and COPY_TARGET with "/opt/airflow"
+# this will let us build the prod image from sources rather than
+# from remote URL from GitHub
+ARG COPY_SOURCE=Dockerfile
+ARG COPY_TARGET=/Dockerfile
+
+## Only needed for installing from sources
+COPY ${COPY_SOURCE} ${COPY_TARGET}
```

Comment:
I don't get what this is for? Why do we need the Dockerfile in the image at all?

Reply:
See IMAGES.rst. It's all explained there. I am using the same Dockerfile to build from a) sources or b) from remote URL. The construct that I am using here allows to specify different ways how we get the sources and build them using the same way during the CI (so that we can test the production Docker image from sources) or from the released github version  of ta airflow. This way we can get stable, repeatable release from Github Sources as well as continuous testing of the image during our CI builds.

We are going to use this very image (build from sources) and helm charts donated by Astronomer to run tests of all three together: Dockerfile (image), helm chart and tests of various executors installed with the helm charts.

Reply:
It is? I read it and nothing there told me why we'd want the Dockerfile in the docker image.